src/devices/sunricher.ts

```diff
@@ -25,13 +25,150 @@ import {
+                await entity.write('genBasic', {[attribute]: {value: numericValue, type: dataType}}, {manufacturerCode: sunricherManufacturerCode});
+                return {state: {externalSwitchType: value}};
+            },
+            convertGet: async (entity, key, meta) => {
+                await entity.read('genBasic', [attribute], {manufacturerCode: sunricherManufacturerCode});
+            },
+        } satisfies Tz.Converter,
+    ];
+
+    const exposes: Expose[] = [e.enum('externalSwitchType', ea.ALL, ['Push button', 'Normal on/off', 'Three way']).withLabel('External Switch Type')];
+    const onEvent: OnEvent = async (type, data, device) => {
+        await readAttributeIfOnline(type, data, device, attribute);
+    };
+    return {
+        fromZigbee,
+        toZigbee,
+        exposes,
+        onEvent,
+        isModernExtend: true,
+    };
+}
+
+function sunricherMinimumPWM(): ModernExtend {
+    const attribute = 0x7809;
+    const dataType = 0x20;
+
+    const fromZigbee: Fz.Converter[] = [
+        {
+            cluster: 'genBasic',
+            type: ['attributeReport', 'readResponse'],
+            convert: (model, msg, publish, options, meta) => {
+                if (Object.prototype.hasOwnProperty.call(msg.data, attribute)) {
+                    console.log(`from `, msg.data[attribute]);
+                    const value = Math.round(msg.data[attribute] / 5.1);
+                    return {
+                        minimumPWM: value,
+                    };
+                }
+                return undefined;
+            },
+        },
+    ];
+
+    const toZigbee: Tz.Converter[] = [
+        {
+            key: ['minimumPWM'],
+            convertSet: async (entity: Zh.Endpoint, key: string, value: number | string, meta) => {
+                console.log(`to `, value);
+                const numValue = typeof value === 'string' ? parseInt(value) : value;
+                const zgValue = Math.round(numValue * 5.1);
+                await entity.write('genBasic', {[attribute]: {value: zgValue, type: dataType}}, {manufacturerCode: sunricherManufacturerCode});
+                return {state: {minimumPWM: numValue}};
+            },
+            convertGet: async (entity: Zh.Endpoint, key: string, meta) => {
+                await entity.read('genBasic', [attribute], {manufacturerCode: sunricherManufacturerCode});
+            },
+        },
+    ];
+
+    const exposes: Expose[] = [
+        e
+            .numeric('minimumPWM', ea.ALL)
+            .withLabel('Minimum PWM')
+            .withDescription('Power off the device and wait for 3 seconds before reconnecting to apply the settings.')
+            .withValueMin(0)
+            .withValueMax(50)
+            .withUnit('%')
+            .withValueStep(1),
+    ];
+
+    const onEvent: OnEvent = async (type, data, device) => {
+        await readAttributeIfOnline(type, data, device, attribute);
```

Comment:
I think these only  need to be read one time after the device has joined? If yes, put them in a `configure`.